dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/jaxb/JAXBContextCache.java

```diff
@@ -0,0 +1,474 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cxf.common.jaxb;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.annotation.XmlElementDecl;
+import javax.xml.transform.dom.DOMSource;
+
+import org.apache.cxf.common.classloader.ClassLoaderUtils;
+import org.apache.cxf.common.util.CacheMap;
+import org.apache.cxf.common.util.CachedClass;
+import org.apache.cxf.common.util.StringUtils;
+import org.apache.cxf.common.logging.LogUtils;
+
+/**
+ *
+ */
+public final class JAXBContextCache {
+
+    /**
+     * Return holder of the context, classes, etc...
+     * Do NOT hold onto these strongly as that can lock the JAXBContext and Set<Class> objects
+     * into memory.  It preferred to grab the context and classes (if needed) from this object
+     * immediately after the call to getCachedContextAndSchemas and then discard it.  The
+     * main purpose of this class is to hold onto the context/set strongly until the caller
+     * has a chance to copy those into a place where they can hold onto it strongly as
+     * needed.
+     */
+    private static final Logger LOG = LogUtils.getLogger(JAXBContextCache.class);  // Liberty Change start
+    private static boolean isLoggableFine = LOG.isLoggable(Level.FINE);
```

Comment:
Should this be `final`?

Reply:
On further thought, I don't think we should cache this check at the class level. If we do, we can't dynamically update the trace without restarting the server/application.

You could cache it at the start of a trace heavy method if performance is concerned.